docs/develop/go/converters-and-encryption.mdx

```diff
@@ -18,11 +18,12 @@ tags:
   - Temporal SDKs
 ---
 
-This page shows how to do the following:
+By default, Temporal payloads are stored unencrypted inside of its data store.
```

Comment:
I'm a bit uncomfortable with how this idea is introduced. Nothing strictly wrong from a technical perspective, but I think we can give a much more positive feel to the subject.

- Starting this page with "By default" gives a first impression that this is some kind of option you need to turn on or configure. Then the rest of the section clarifies the idea, but at that point, the reader is left with the feeling that encryption is a missing/incomplete feature in Temporal, whereas it is that way by-design…

- From the Temporal Server's perspective, payloads are ALWAYS stored "unencrypted" in the data store (i.e. the server itself never add an extra layer of encryption; the data is stored as it received it). The Server don't even know either your data is encrypted or not.

I'd suggest introducing the idea from the following angle instead:

- Temporal's security model is designed around client-side encryption of Payloads, i.e. a client may encrypt Payloads _before_ sending them to the server (and decrypt them _after_ receiving them from the server). This provides a very high degree of confidentiality because the Temporal Server itself has absolutely no knowledge of the actual data. It also gives implementers more power and more freedom regarding which client is able to read which data (control access on keys, algorithms, etc).

- A Temporal developer adds client-side encryption of Payloads by providing a Custom Payload Codec to its Client. Depending on business needs, a complete implementation of Payload Encryption may involve selection of appropriate encryption algorithms, management of encryption keys, or restrict a subset of their users from viewing payload output. (etc)

- The server itself never adds encryption over Payloads. Therefore, unless client-side encryption is implemented, Payload data will be persisted in non-encrypted form to the data store, and any Client that can make requests to a Temporal namespace (including the Temporal UI and CLI) will be able to read Payloads contained in Workflows. When working with sensitive data, one should ALWAYS implement Payload encryption.

Reply:
I like this a lot and I'm using it nearly verbatim, thanks!